commands/cli/parse.go

```diff
@@ -337,6 +317,10 @@ func parseArgs(inputs []string, stdin *os.File, argDefs []cmds.Argument, recursi
 			} else {
 				if stdin != nil && argDef.SupportsStdin &&
 					argDef.Required && !fillingVariadic {
+					stdInfo, _ := stdin.Stat()
```

Comment:
Same here, maybe extract it to function "isTerminal" or "printReadInfo" so it doesn't repeat.